### ISSUE TITLE
Update workflows that make commits to use bot

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#6118]
## Feature Description

Google's CLA requirement for code contributions applies to all commits by default. For commits authored by a bot, an exception can be made for Google-owned entities. In #5345 we introduced our first custom workflow which creates maintenance PRs for us, but this will need to be updated to author commits using the new aforementioned bot.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* All workflows that make programmatic commits to this repository, should be updated to do so using the new bot user: `google-site-kit-bot`
* Git commits by the bot should use the following config
	* The committer and author names should use the bot's username: `google-site-kit-bot`
	* The committer and author emails should use the bot's noreply email: `138160377+google-site-kit-bot@users.noreply.github.com`
* Commits should be authenticated as the bot user using the `GH_BOT_TOKEN` secret

_See https://github.com/GoogleForCreators/web-stories-wp/blob/c69c571edc5976e13f3221e65b14aa3079d7e967/.github/workflows/update-browserslist.yml for a good example_

## Implementation Brief

* Follow the example linked in the AC to update the following workflows with configuration to set the git config as per the AC and checkout the repository that will be committed to using the bot's token secret
	* `browserslist-db-update.yml`
	* `storybook.yml` (deploy)

### Test Coverage

* No additional coverage is needed because GitHub Action workflows don't have their own tests 

## QA Brief

* Git commits to Storybook and Browserlist should be made using the `google-site-kit-bot` account.

## Changelog entry

* N/A.

**Comment by bethanylang** [comment]
@felixarntz Checking in here, thanks!

**Comment by felixarntz** [comment]
@bethanylang This is relatively low priority and blocked by a Googler setting up the bot, I'll create an Asana task for myself, but may take a while.

**Comment by bethanylang** [comment]
@felixarntz Thanks for the update!

**Comment by techanvil** [comment]
IB :white_check_mark:

**Comment by kuasha420** [comment]
# QA:Eng :x: 

* :heavy_check_mark: Commits in [storybook](https://github.com/google/site-kit-wp/commit/b3c2797d139e3aa7d7dd630bd1ebe49d013beaad) and [browserlist-db](https://github.com/google/site-kit-wp/pull/7502/commits/fbe7bf1f3dc45ef24343de1d51f6218c5426bced) PR are being created by `google-site-kit-bot`.
* :x: The [PR](https://github.com/google/site-kit-wp/pull/7502) for updating browserlist-db is still being created by `github-actions` bot instead of `google-site-kit-bot` which is not the case for the [web-stories](https://github.com/GoogleForCreators/web-stories-wp/pull/13433) repo.
* :warning: The CLA check is still failing in the PR, is this expected? 

cc @aaemnnosttv @wpdarren

**Comment by tofumatt** [comment]
Odd, the CLA check didn't fail my first time running it 🤔 

Looks like the Browserlist _commit_ is being created by the Site Kit bot:

<img width="364" alt="CleanShot 2023-09-04 at 09 51 07" src="https://github.com/google/site-kit-wp/assets/90871/450d2bb4-3ccf-49a4-a995-8fdfacfd06ad">

But not the PR itself:

<img width="740" alt="CleanShot 2023-09-04 at 09 51 44" src="https://github.com/google/site-kit-wp/assets/90871/32faa30a-ed0a-48f1-88cd-6c08f499b97a">

That's probably what's causing that CLA check to fail, because of the mismatch between PR creator and commits, but also because the GitHub Actions bot hasn't signed the CLA.

I'll take a look and see if we can have the PR itself be made by the Site Kit bot.

**Comment by tofumatt** [comment]
Alright, I've updated everything to use the access token for the Site Kit bot, which seems like it's already been set up.

Looks like we need to add the Site Kit bot to the list of CLA users… I have no idea how since it uses a no-reply email. 🤔 

https://github.com/google/site-kit-wp/pull/7540/checks?check_run_id=16472791026

@aaemnnosttv: are you able to manually add that email to the CLA list? Or maybe @swissspidy knows how as they seem to have it set up on other repos.

**Comment by swissspidy** [comment]
Hi there 👋 

From what I can see the bot has been registered correctly already in [b/289425301](http://b/289425301) (Google-internal link)

The issue is that #7539 merely changes the token that's used for opening the PR, but it does not actually perform the _commit_ as the bot.

If you take a look at #7540, the commit author is `google-site-kit-bot <138160377+google-site-kit-bot@users.noreply.github.com>`. This needs to be the bot's email address.

Please consult the [documentation](https://github.com/c2corg/browserslist-update-action) of the `browserslist-update-action` GitHub action you're using, it shows how to properly configure the email address using `git config --global` You can also accomplish this using environment variables, as we do in the Web Stories repository: https://github.com/GoogleForCreators/web-stories-wp/blob/1ca31eb1d739e67463abd8921cc00ddc1fc48a4c/.github/workflows/update-browserslist.yml#L11-L15

If it still doesn't work, have @aaemnnosttv go through the checklist at go/github-docs/robots again. I'll be on vacation soon, so can't help further.

**Comment by tofumatt** [comment]
@aaemnnosttv Looks like you might need to run through the checklist above and add the bot to the CLA list manually, as otherwise everything here is sorted. I'm going to assign this one to you to make that change/review the PR.

**Comment by swissspidy** [comment]
Have you changed the Git email accordingly? I don‘t see that change in #7539

**Comment by tofumatt** [comment]
@swissspidy Sorry, that PR isn't the only one that was made. There was also #7455, which already set the user. It set the GitHub user appropriately.

**Comment by swissspidy** [comment]
That doesn‘t use the bots Google email address though

**Comment by tofumatt** [comment]
The PR here is one generated by this updated action: #7553.

Both the commit and PR author/committer are the appropriate user (3f61ea1263189566f24aa11dac2c4a18c0a87bc5), but the CLA is unsigned.

@aaemnnosttv I think you'll need to get it added to the CLA list. The user is this one: @google-site-kit-bot

**Comment by tofumatt** [comment]
> That doesn‘t use the bots Google email address though

@swissspidy Ah, thanks, looks like it wasn't supplied in this issue. I can't see the email address of that user, could you or @aaemnnosttv supply that address here and then I can get it added? 🙂

**Comment by aaemnnosttv** [comment]
Thanks @swissspidy !

> That doesn‘t use the bots Google email address though

Right, I thought that was the point of using the noreply address, just like it is used in web-stories as linked above https://github.com/GoogleForCreators/web-stories-wp/blob/1ca31eb1d739e67463abd8921cc00ddc1fc48a4c/.github/workflows/update-browserslist.yml#L11-L15 which passes the CLA check https://github.com/GoogleForCreators/web-stories-wp/runs/16416022483

From the failing check, it seems to be possible to add this address to the CLA as well so I'll inquire internally to see if we can do that first. If not, we can follow-up to update the address.

The browserslist db action is the first that involved creating PRs with the bot so we can simply hold on merging any of those as long as this check is failing (as we have been for a while now).

Since the fix at this point is likely external, this need not block this issue from being completed.

**Comment by aaemnnosttv** [comment]
Closing as this is being actioned externally. Will follow-up with separate issue if needed 👍
